ui/pages/asset/components/token-buttons.tsx

```diff
@@ -123,14 +131,34 @@ const TokenButtons = ({
   }, [token.isERC721, token.address, dispatch]);
 
   const setCorrectChain = async () => {
+    // If we aren't presently on the chain of the asset, change to it
     if (currentChainId !== token.chainId) {
-      const networkConfigurationId = networks[token.chainId];
-      await dispatch(setActiveNetwork(networkConfigurationId));
-      await dispatch(
-        setSwitchedNetworkDetails({
-          networkClientId: networkConfigurationId,
-        }),
-      );
+      try {
+        const networkConfigurationId = networks[token.chainId];
+        await dispatch(setActiveNetwork(networkConfigurationId));
+        await dispatch(
```

Comment:
I see that the behavior when this fails is that if the user initially wanted to send BUSD on polygon while he is on Ethereum network; he will end-up seeing a screen with BUSD on ethereum (i think even sometimes we have case where token does not exist in network but we will still show the token logo with another network logo)

The user wont be able to eventually send the amount of token he intended because the UI wont allow it, but nothing prevents him from clicking next and confirming a tx where he will end up paying the fees for;

https://github.com/user-attachments/assets/34dfb5da-b185-4977-9cf1-6ecdd9a104a1

Reply:
Per our conversation, we are going to add additional conditions to return early when there is a mismatched chainId between tokens and networks.

This can sometimes result in a user needing to double click the send/swap button. Maybe we can revisit this post launch for a more elegant solution.